math/src/polynomial.rs

```diff
@@ -35,6 +35,9 @@ impl<F: IsField> Polynomial<FieldElement<F>> {
         Self::new(&[])
     }
 
+    /// Returns a polynomial that interpolates the points with x coordinates and y coordinates given by
+    /// `xs` and `ys`.
+    /// `xs` and `ys` must be the same length, and `xs` values should be unique. If not, panics.
     pub fn interpolate(xs: &[FieldElement<F>], ys: &[FieldElement<F>]) -> Self {
         // TODO: try to use the type system to avoid this assert
```

Comment:
If by chance the size is compile-time constant, we could probably do:
```suggestion
    pub fn interpolate<const N: usize>(xs: &[FieldElement<F>; N], ys: &[FieldElement<F>; N]) -> Self {
        // TODO: try to use the type system to avoid this assert
```

Reply:
Sadly, it is not 😞

I was thinking maybe unify them to `&[[FieldElement<F>; 2]]`, `&[(FieldElement<F>, FieldElement<F>)]`, some wrapper struct, or an iterator that returns tuples of x and y values.